File: Linnworks/src/net/LinnworksAPI/PurchaseOrder.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace LinnworksAPI
{
    public static class PurchaseOrderMethods
    {
        private static JsonSerializerSettings serializerSettings = new JsonSerializerSettings() { DateFormatString = "yyyy-MM-ddTHH:mm:ss.ffZ" };

        public static Search_PurchaseOrdersResult Search_PurchaseOrders(Search_PurchaseOrderParameter searchParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Search_PurchaseOrdersResult>(Factory.GetResponse("PurchaseOrder/Search_PurchaseOrders", "searchParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(searchParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Guid Create_PurchaseOrder_Initial(Create_PurchaseOrder_InitialParameter createParameters, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Guid>(Factory.GetResponse("PurchaseOrder/Create_PurchaseOrder_Initial", "createParameters=" + Newtonsoft.Json.JsonConvert.SerializeObject(createParameters, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static PurchaseOrderHeader Update_PurchaseOrderHeader(Update_PurchaseOrderHeaderParameter updateParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<PurchaseOrderHeader>(Factory.GetResponse("PurchaseOrder/Update_PurchaseOrderHeader", "updateParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(updateParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Update_PurchaseOrderItemResponse Add_PurchaseOrderItem(Add_PurchaseOrderItemParameter addItemParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Update_PurchaseOrderItemResponse>(Factory.GetResponse("PurchaseOrder/Add_PurchaseOrderItem", "addItemParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(addItemParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Update_PurchaseOrderItemResponse Update_PurchaseOrderItem(Update_PurchaseOrderItemParameter updateItemParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Update_PurchaseOrderItemResponse>(Factory.GetResponse("PurchaseOrder/Update_PurchaseOrderItem", "updateItemParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(updateItemParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static PurchaseOrderHeader Delete_PurchaseOrderItem(Delete_PurchaseOrderItemParameter deleteItemParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<PurchaseOrderHeader>(Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrderItem", "deleteItemParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(deleteItemParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<Guid> GetPurchaseOrdersWithStockItems(PurchaseOrderWithStockItem purchaseOrder, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<Guid>>(Factory.GetResponse("PurchaseOrder/GetPurchaseOrdersWithStockItems", "purchaseOrder=" + Newtonsoft.Json.JsonConvert.SerializeObject(purchaseOrder, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Get_PurchaseOrderResponse Get_PurchaseOrder(Guid pkPurchaseId, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Get_PurchaseOrderResponse>(Factory.GetResponse("PurchaseOrder/Get_PurchaseOrder", "pkPurchaseId=" + pkPurchaseId + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<PurchaseOrderNote> Get_PurchaseOrderNote(Guid pkPurchaseId, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<PurchaseOrderNote>>(Factory.GetResponse("PurchaseOrder/Get_PurchaseOrderNote", "pkPurchaseId=" + pkPurchaseId + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static PurchaseOrderNote Add_PurchaseOrderNote(Guid pkPurchaseId, String Note, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<PurchaseOrderNote>(Factory.GetResponse("PurchaseOrder/Add_PurchaseOrderNote", "pkPurchaseId=" + pkPurchaseId + "&Note=" + Note + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Int32 Delete_PurchaseOrderNote(Guid pkPurchaseId, Guid pkPurchaseOrderNoteId, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Int32>(Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrderNote", "pkPurchaseId=" + pkPurchaseId + "&pkPurchaseOrderNoteId=" + pkPurchaseOrderNoteId + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static PurchaseOrderHeader Change_PurchaseOrderStatus(Change_PurchaseOrderStatusParameter changeStatusParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<PurchaseOrderHeader>(Factory.GetResponse("PurchaseOrder/Change_PurchaseOrderStatus", "changeStatusParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(changeStatusParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Deliver_PurchaseOrderItemAllResponse Deliver_PurchaseItemAll(Guid purchaseId, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Deliver_PurchaseOrderItemAllResponse>(Factory.GetResponse("PurchaseOrder/Deliver_PurchaseItemAll", "purchaseId=" + purchaseId + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static Update_PurchaseOrderItemResponse Deliver_PurchaseItem(Deliver_PurchaseItemParameter deliverItemParameter, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Update_PurchaseOrderItemResponse>(Factory.GetResponse("PurchaseOrder/Deliver_PurchaseItem", "deliverItemParameter=" + Newtonsoft.Json.JsonConvert.SerializeObject(deliverItemParameter, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<PurchaseOrderStatus> GetPurchaseOrderStatusList(String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<PurchaseOrderStatus>>(Factory.GetResponse("PurchaseOrder/GetPurchaseOrderStatusList", "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static void Delete_PurchaseOrder(Guid pkPurchaseId, String ApiToken, String ApiServer)
        {
            Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrder", "pkPurchaseId=" + pkPurchaseId + "", ApiToken, ApiServer);
        }

        public static List<PurchaseOrderAudit> Get_PurchaseOrderAudit(Search_PurchaseOrder_AuditLog auditLog, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<PurchaseOrderAudit>>(Factory.GetResponse("PurchaseOrder/Get_PurchaseOrderAudit", "auditLog=" + Newtonsoft.Json.JsonConvert.SerializeObject(auditLog, serializerSettings) + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }
    }
}

File: Linnworks/src/javascript/PurchaseOrder.js
var PurchaseOrder =
{
	// http://apidoc.linnworks.net/#/PurchaseOrder-Add_PurchaseOrderExtendedProperty
	Add_PurchaseOrderExtendedProperty: function(request,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Add_PurchaseOrderExtendedProperty", token, server, "request=" + JSON.stringify(request) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Add_PurchaseOrderItem
	Add_PurchaseOrderItem: function(addItemParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Add_PurchaseOrderItem", token, server, "addItemParameter=" + JSON.stringify(addItemParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Add_PurchaseOrderNote
	Add_PurchaseOrderNote: function(pkPurchaseId,Note,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Add_PurchaseOrderNote", token, server, "pkPurchaseId=" + pkPurchaseId + "&Note=" + Note +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Change_PurchaseOrderStatus
	Change_PurchaseOrderStatus: function(changeStatusParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Change_PurchaseOrderStatus", token, server, "changeStatusParameter=" + JSON.stringify(changeStatusParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Create_PurchaseOrder_Initial
	Create_PurchaseOrder_Initial: function(createParameters,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Create_PurchaseOrder_Initial", token, server, "createParameters=" + JSON.stringify(createParameters) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Delete_PurchaseOrder
	Delete_PurchaseOrder: function(pkPurchaseId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrder", token, server, "pkPurchaseId=" + pkPurchaseId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Delete_PurchaseOrderExtendedProperty
	Delete_PurchaseOrderExtendedProperty: function(request,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrderExtendedProperty", token, server, "request=" + JSON.stringify(request) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Delete_PurchaseOrderItem
	Delete_PurchaseOrderItem: function(deleteItemParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrderItem", token, server, "deleteItemParameter=" + JSON.stringify(deleteItemParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Delete_PurchaseOrderNote
	Delete_PurchaseOrderNote: function(pkPurchaseId,pkPurchaseOrderNoteId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Delete_PurchaseOrderNote", token, server, "pkPurchaseId=" + pkPurchaseId + "&pkPurchaseOrderNoteId=" + pkPurchaseOrderNoteId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Deliver_PurchaseItem
	Deliver_PurchaseItem: function(deliverItemParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Deliver_PurchaseItem", token, server, "deliverItemParameter=" + JSON.stringify(deliverItemParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Deliver_PurchaseItemAll
	Deliver_PurchaseItemAll: function(purchaseId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Deliver_PurchaseItemAll", token, server, "purchaseId=" + purchaseId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Deliver_PurchaseItemAll_ExceptBatchItems
	Deliver_PurchaseItemAll_ExceptBatchItems: function(purchaseId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Deliver_PurchaseItemAll_ExceptBatchItems", token, server, "purchaseId=" + purchaseId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Get_PurchaseOrder
	Get_PurchaseOrder: function(pkPurchaseId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Get_PurchaseOrder", token, server, "pkPurchaseId=" + pkPurchaseId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Get_PurchaseOrderAudit
	Get_PurchaseOrderAudit: function(auditLog,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Get_PurchaseOrderAudit", token, server, "auditLog=" + JSON.stringify(auditLog) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Get_PurchaseOrderExtendedProperty
	Get_PurchaseOrderExtendedProperty: function(request,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Get_PurchaseOrderExtendedProperty", token, server, "request=" + JSON.stringify(request) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Get_PurchaseOrderNote
	Get_PurchaseOrderNote: function(pkPurchaseId,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Get_PurchaseOrderNote", token, server, "pkPurchaseId=" + pkPurchaseId +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-GetPurchaseOrderStatusList
	GetPurchaseOrderStatusList: function(token, server)
	{
		return Factory.GetResponse("PurchaseOrder/GetPurchaseOrderStatusList", token, server, "");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-GetPurchaseOrdersWithStockItems
	GetPurchaseOrdersWithStockItems: function(purchaseOrder,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/GetPurchaseOrdersWithStockItems", token, server, "purchaseOrder=" + JSON.stringify(purchaseOrder) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Search_PurchaseOrders
	Search_PurchaseOrders: function(searchParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Search_PurchaseOrders", token, server, "searchParameter=" + JSON.stringify(searchParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Update_PurchaseOrderExtendedProperty
	Update_PurchaseOrderExtendedProperty: function(request,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Update_PurchaseOrderExtendedProperty", token, server, "request=" + JSON.stringify(request) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Update_PurchaseOrderHeader
	Update_PurchaseOrderHeader: function(updateParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Update_PurchaseOrderHeader", token, server, "updateParameter=" + JSON.stringify(updateParameter) +"");
	},
	// http://apidoc.linnworks.net/#/PurchaseOrder-Update_PurchaseOrderItem
	Update_PurchaseOrderItem: function(updateItemParameter,token, server)
	{
		return Factory.GetResponse("PurchaseOrder/Update_PurchaseOrderItem", token, server, "updateItemParameter=" + JSON.stringify(updateItemParameter) +"");
	},
};
